### DISCUSSION TITLE
Working with .bib files autocomplete references using @ and tab

### DISCUSSION THREAD

**Discussion by victorfeagins** [quarto-dev/quarto-cli#2441] [Q&A] [answered]
In Rstudio I was utilizing a .bib file to use references in my document. I noticed that \@ symbol is not able to detect the references inside my .bib file specified in my YAML and I have to type them in manually.

Overleaf has this feature when using \ref{} and thought it might be a nice addition.

**Comment by dragonstyle** [comment, marked as answer]
Rstudio does support a visual mode which does resolve / suggest `@` using the bibliography (it will also resolve cross references). There is some other special sauce like a pretty robust ability to insert citations from other sources (Zotero libraries, DOI, etc...) which automates the addition of the citation to your bibliography file. 

More here:
https://www.rstudio.com/blog/exploring-rstudio-visual-markdown-editor/

And here:
https://www.rstudio.com/blog/rstudio-1-4-preview-citations/

**Reply by victorfeagins** [reply]
Wow yep you are correct. I prefer to working in the source mode, so I didn't see it.
